src/vcpkg/binarycaching.cpp

```diff
@@ -1255,6 +1255,159 @@ namespace
         Path m_tool;
     };
 
+    struct AzureUpkgTool
+    {
+        AzureUpkgTool(const ToolCache& cache, MessageSink& sink) { az_cli = cache.get_tool_path(Tools::AZCLI, sink); }
+
+        Command base_cmd(const AzureUpkgSource& src,
+                         StringView package_name,
+                         StringView package_version,
+                         StringView az_cmd) const
+        {
+            Command cmd{az_cli};
+            cmd.string_arg("artifacts").string_arg("universal").string_arg(az_cmd);
+            cmd.string_arg("--organization").string_arg(src.organization);
+            cmd.string_arg("--feed").string_arg(src.feed);
+            cmd.string_arg("--name").string_arg(package_name);
+            cmd.string_arg("--version").string_arg(package_version);
+            if (!src.project.empty())
+            {
+                cmd.string_arg("--project").string_arg(src.project);
+                cmd.string_arg("--scope").string_arg("project");
+            }
+            return cmd;
+        }
+
+        ExpectedL<Unit> download(const AzureUpkgSource& src,
+                                 StringView package_name,
+                                 StringView package_version,
+                                 const Path& download_path,
+                                 MessageSink& sink) const
+        {
+            Command cmd = base_cmd(src, package_name, package_version, "download");
+            cmd.string_arg("--path").string_arg(download_path);
+            return run_az_artifacts_cmd(cmd, sink);
+        }
+
+        ExpectedL<Unit> publish(const AzureUpkgSource& src,
+                                StringView package_name,
+                                StringView package_version,
+                                const Path& package_dir,
+                                StringView description,
+                                MessageSink& sink) const
+        {
+            Command cmd = base_cmd(src, package_name, package_version, "publish");
+            cmd.string_arg("--description").string_arg(description);
+            cmd.string_arg("--path").string_arg(package_dir);
+            return run_az_artifacts_cmd(cmd, sink);
+        }
+
+        ExpectedL<Unit> run_az_artifacts_cmd(const Command& cmd, MessageSink& sink) const
+        {
+            RedirectedProcessLaunchSettings show_in_debug_settings;
+            show_in_debug_settings.echo_in_debug = EchoInDebug::Show;
+            return cmd_execute_and_capture_output(cmd, show_in_debug_settings)
+                .then([&](ExitCodeAndOutput&& res) -> ExpectedL<Unit> {
+                    if (res.exit_code == 0)
+                    {
+                        return {Unit{}};
+                    }
+                    if (res.output.find("you need to run the login command") != std::string::npos)
```

Comment:
Can you describe in a comment

1. How you experienced the tool emitting this output? (That is, how can a code reviewer trying to do this themselves recreate what this find is looking for)
2. What the full output is supposed to look like that we are parsing here?
3. Do we know if this works on systems with a language other than English set as the default?

Reply:
Added a comment, not sure what is displayed on systems in a language other than English